### ISSUE TITLE
sysrepo: Validation of the changes failed

### ISSUE THREAD

**Issue by Rohit2387** [CESNET/Netopeer2#239]
Hi,

Below error is coming when connect request is send from netopper2-cli.However connect was succesfull.
netopeer2-server[19781]: np2srv_sr_event_notif_send failed (sysrepo: Validation of the changes failed) 
Can you please help me know the reason of this issue?

Br
Rohit

**Comment by michalvasko** [comment]
Hi Rohit.
please run *sysrepo* as `# sysrepo -d -l3` and provide output after you get this error in *netopeer2-server*.

Regards,
Michal

**Comment by Rohit2387** [comment]
Hi,
sysrepo is not present sysrepod is there,is it due to that?

Br
Rohit

**Comment by Rohit2387** [comment]
And also below sysrepo make install is failing in import.
 Exec: /root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/ietf-interfaces@2014-05-08.yang --permissions=666; /root/sysrepo-master/build/src/sysrepocfg --import=/root/sysrepo-master/examples/yang/ietf-interfaces.data.xml --datastore=startup --format=xml --level=0 ietf-interfaces
CMake Error at examples/cmake_install.cmake:63 (message):
  Error: 1
Call Stack (most recent call first):
  cmake_install.cmake:162 (INCLUDE)

**Comment by michalvasko** [comment]
Hi,
sorry, I meant `sysrepod`, of course.

That last error is definitely a problem and I would solve that one first. It would be also better if you used `devel` but installation should work in `master` as well. Could you paste the rest of the error, the included part is not too helpful.

Regards,
Michal

**Comment by Rohit2387** [comment]
Exec: /root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/turing-machine.yang --permissions=666; /root/sysrepo-master/build/src/sysrepocfg --import=/root/sysrepo-master/examples/yang/turing-machine.data.xml --datastore=startup --format=xml --level=0 turing-machine
-- Exec: /root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/iana-if-type.yang --permissions=666
-- Exec: /root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/ietf-ip@2014-06-16.yang --permissions=666
-- Exec: /root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/ietf-interfaces@2014-05-08.yang --permissions=666; /root/sysrepo-master/build/src/sysrepocfg --import=/root/sysrepo-master/examples/yang/ietf-interfaces.data.xml --datastore=startup --format=xml --level=0 ietf-interfaces
CMake Error at examples/cmake_install.cmake:63 (message):
  Error: 1
Call Stack (most recent call first):
  cmake_install.cmake:162 (INCLUDE)


make: *** [install] Error 1

**Comment by michalvasko** [comment]
Um, I have no idea what that means, there is no actual error message even though I think there should be. Can you maybe try executing those commands manually and with increased verbosity? Such as
```
/root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/ietf-interfaces@2014-05-08.yang --permissions=666; /root/sysrepo-master/build/src/sysrepocfg --import=/root/sysrepo-master/examples/yang/ietf-interfaces.data.xml --datastore=startup --format=xml --level=3 ietf-interfaces
```

Regards,
Michal

**Comment by Rohit2387** [comment]
And also below command return failure:-
sysrepocfg -d running ietf-interfaces -x
Cannot operate on the running datastore for 'ietf-interfaces' as there are no active subscriptions for it.
Canceling the operation.

**Comment by michalvasko** [comment]
I was just talking about that in another issue, please refer the [first paragraph](https://github.com/sysrepo/sysrepo/issues/1103).

**Comment by Rohit2387** [comment]
Still same error after ruining mentioned command :(
netopeer2-server[10261]: np2srv_sr_event_notif_send failed (sysrepo: Validation of the changes failed).
By the way sysrepo is needed for netopper2?

**Comment by michalvasko** [comment]
Yes, *sysrepo* is needed for *netopeer2* and please stop jumping from one problem to another. Did you do what I asked you to? What was the output of
```
/root/sysrepo-master/build/src/sysrepoctl --install --yang=/root/sysrepo-master/examples/yang/ietf-interfaces@2014-05-08.yang --permissions=666; /root/sysrepo-master/build/src/sysrepocfg --import=/root/sysrepo-master/examples/yang/ietf-interfaces.data.xml --datastore=startup --format=xml --level=3 ietf-interfaces
```

**Comment by Rohit2387** [comment]
yes command was successful.Yes but library will be installed in /usr/lib right?

**Comment by michalvasko** [comment]
Right, so I don't know what distribution are you using but you probably need to compile `libpcre` yourself with support for the mentioned options.

**Comment by Rohit2387** [comment]
Yes same thing i was trying i recompiled libpcre with below option(`--enable-utf` and `--enable-unicode-properties) and then libyang and then sysrepo.It works thanks.
